.eslintrc.js

```diff
@@ -222,9 +136,30 @@ module.exports = {
         'import/extensions': ['error', 'ignorePackages'],
       },
     },
+
+    // Rules only impacting bundled files and doc files
+    {
+      files: ['packages/*/src/**/*{.ts,.tsx,.js}', 'docs/data/**/*{.ts,.tsx,.js}'],
```

Comment:
I suspect that we don't need this. This should be coming from the monorepo root eslint configuration since there doesn't seem to be anything specific about MUI X.

```suggestion
```